fix(widget): refactor plugin for canary deploy

E-Rick opened the pull request:
### All Submissions:

* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

Fixes the constructed data attributes in the recent posts function and recommendation function.

Also fixes the nesting of the recommendations and recent posts unordered lists so that the AMP experiments can select them correctly.

### How to test the changes in this Pull Request:

1. Build the plugin and upload to WP
2. Set the widget in sidebar from widget settings
3. Open a post
4. You should see article recommendations and recent posts both displayed.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

E-Rick committed:
fix: add data-vars to recent and rec helpers

E-Rick committed:
feat: add data-vars-click-url attributes

tngzng commented on article-recommendations-widget.php:
```diff
@@ -69,14 +110,13 @@ public function widget($args, $instance) {
 		$sort_by = '&sort_by=score';
 
 		// This is where you run the code and display the output
 		$dev_url = "https://dev-article-rec-api.localnewslab.io/recs";
```
any reason not to use the prod_url at this point?

tngzng commented on article-recommendations-widget.php:
```diff
@@ -92,31 +132,77 @@ public function widget($args, $instance) {
 		 * @return void
 		 */
 		function lnl_get_recommendations($results) {
+			$id = $results[0]['model']['id'];
+			$type = $results[0]['model']['type'];
+			$status = $results[0]['model']['status'];
```
just for clarity can we prefix each of these variables with `model_`?

tngzng commented on article-recommendations-widget.php:
```diff
@@ -146,34 +230,48 @@ function lnl_get_recommendations($results) {
+			$str .= "data-vars-three=$three ";
+			$str .= "data-vars-four=$four ";
+			$str .= "data-vars-five=$five ";
+			$str .= "data-vars-model-type='recent_posts' ";
```
should we interpolate `model_type` into the string, rather than hardcoding recent_posts?

tngzng commented on article-recommendations-widget.php:
```diff
@@ -92,31 +132,77 @@ public function widget($args, $instance) {
+			$str .= "data-vars-four=$four ";
+			$str .= "data-vars-five=$five ";
+
+			echo "<ul id='recommendations' " . $str . ">";
```
since echoing has been iffy for us, does it make sense for us to build a string of html in `lnl_get_recommendations` and `lnl_get_recent_posts`, return that from each and then echo those where the functions are called, ie:
```
$recs = lnl_get_recommendations($results);		
echo $recs;		
$recent_posts = lnl_get_recent_posts();		
echo $recent_posts;
```
^ idk php so that syntax could very well be off

E-Rick replied:
I'm not sure, could try to fix in next iteration

tngzng reviewed (changes requested):
just a couple comments around variable naming and comments! also curious what you think about changing the return values of those funcs from void to strings

E-Rick committed:
fix: add end <ul> tag on request error

E-Rick committed:
refactor: change request url to prod

remove hardcoded 'recent posts' model_type

E-Rick committed:
docs(callback): add func def

E-Rick committed:
chore: update package-lock

E-Rick committed:
revert: remove npm-force-resolutions

E-Rick committed:
refactor(article-recommendations): remove gutenberg plugin

E-Rick committed:
refactor: remove 'widget' from plugin name

E-Rick committed:
chore: change package name to just article-recommendations

E-Rick committed:
refactor(article-recommendations): remove gutenberg src index file

E-Rick committed:
build: remove dist from build script

E-Rick committed:
feat: exclude the current post

E-Rick committed:
refactor: put display posts logic in if-else

E-Rick committed:
fix(rec-api): add NaN check and exclude articles scored NaN

hardcode exclude params for problem articles and add is_nan check in recs function

E-Rick committed:
refactor(get_recommendations): return string and clean up

move function to method, clean up code to wp standards, change return type

E-Rick committed:
refactor: remove lnl prefix from methods

* style: fix brace styles

E-Rick committed:
refactor(get_recent_posts): change return type and cleanup

E-Rick committed:
style: fix white-spacing

E-Rick committed:
wip: consolidate both helpers into one function

E-Rick committed:
build: add db user and name to wordpress container

E-Rick committed:
build: fix wp-content volume source path

E-Rick committed:
fix: add exclude again, nullish checks, and defaults

E-Rick reviewed (commented):
Made all the changes Ting proposed as well as other refactors.

The code is more maintainable now, I separated out the functions, added exclude Parameters, and conditionals to display recent_posts when the API fails.

I also added nullish checks on the variables for when I merge the two functions into one in a future PR. The two functions share a lot of the same logic except for the difference in naming conventions from API and WP.

I want to merge these so I can make smaller PRs going forward.

E-Rick committed:
fix: remove widget file
